astropy/modeling/functional_models.py

```diff
@@ -3361,6 +3361,207 @@ def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
+
+        fig, ax = plt.subplots()
+        im = ax.imshow(log_img, origin='lower', interpolation='nearest',
+                       vmin=-1, vmax=2)
+        cbar = fig.colorbar(im, ax=ax)
+        cbar.set_label('Log Brightness', rotation=270, labelpad=25)
+        cbar.set_ticks([-1, 0, 1, 2])
+        plt.title('Boxy isophote with c=1.0')
+        plt.xlabel('x')
+        plt.ylabel('y')
+        plt.show()
+
+    References
+    ----------
+    .. [1] http://ned.ipac.caltech.edu/level5/March05/Graham/Graham2.html
+    .. [2] https://ui.adsabs.harvard.edu/abs/1990MNRAS.245..130A/abstract
+    """
+
+    amplitude = Parameter(default=1, description="Surface brightness at r_eff")
+    r_eff = Parameter(default=1, description="Effective (half-light) radius")
+    n = Parameter(default=4, description="Sersic Index")
+    x_0 = Parameter(default=0, description="X position of the center")
+    y_0 = Parameter(default=0, description="Y position of the center")
+    c = Parameter(default=0, description="Isophote shape parameter")
+    ellip = Parameter(default=0, description="Ellipticity")
+    theta = Parameter(
+        default=0.0,
+        description=(
+            "Rotation angle either as a float (in radians) or a |Quantity| angle"
+        ),
+    )
+    _gammaincinv = None
+
+    @classmethod
+    def evaluate(cls, x, y, amplitude, r_eff, n, x_0, y_0, c, ellip, theta):
+        """Two dimensional Sersic profile function."""
+        if cls._gammaincinv is None:
+            from scipy.special import gammaincinv
+
+            cls._gammaincinv = gammaincinv
+
+        bn = cls._gammaincinv(2.0 * n, 0.5)
+        a, b = r_eff, (1 - ellip) * r_eff
+        cos_theta, sin_theta = np.cos(theta), np.sin(theta)
+        x_maj = np.abs((x - x_0) * cos_theta + (y - y_0) * sin_theta)
+        x_min = np.abs(-(x - x_0) * sin_theta + (y - y_0) * cos_theta)
+
+        expon = 2.0 + c
+        inv_expon = 1.0 / expon
+        z = ((x_maj / a) ** expon + (x_min / b) ** expon) ** inv_expon
+        return amplitude * np.exp(-bn * (z ** (1 / n) - 1.0))
+
+    @property
+    def input_units(self):
+        if self.x_0.input_unit is None:
+            return None
+        return {
+            self.inputs[0]: self.x_0.input_unit,
+            self.inputs[1]: self.y_0.input_unit,
+        }
+
+    def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
+        # Note that here we need to make sure that x and y are in the same
+        # units otherwise this can lead to issues since rotation is not well
+        # defined.
+        if inputs_unit[self.inputs[0]] != inputs_unit[self.inputs[1]]:
+            raise UnitsError("Units of 'x' and 'y' inputs should match")
+        return {
+            "x_0": inputs_unit[self.inputs[0]],
+            "y_0": inputs_unit[self.inputs[0]],
+            "r_eff": inputs_unit[self.inputs[0]],
+            "theta": u.rad,
+            "amplitude": outputs_unit[self.outputs[0]],
+        }
```

Comment:
I believe that `c` also needs to have a unit returned.

Reply:
No, `c` is a unitless exponent.